kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/PaymentRequestDocument.java

```diff
@@ -145,6 +148,15 @@ public void prepareForSave() {
             }
         }
 
+        setPayment1099Indicator(false);
+        for (PaymentRequestIncomeType incomeType : getIncomeTypes()) {
+            boolean isCodeExist = StringUtils.isNotBlank(incomeType.getIncomeTypeCode());
+            boolean isReportable = !incomeType.getIncomeTypeCode().equals(KFSConstants.IncomeTypeConstants.INCOME_TYPE_NON_REPORTABLE_CODE);
+            if (isCodeExist && isReportable) {
+                setPayment1099Indicator(true);
```

Comment:
Same concern as last the for loop I commented on here: isn't it the case that the first positive in the list is all we care about, so why not a break after finding it?